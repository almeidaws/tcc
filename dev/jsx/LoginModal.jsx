import React from 'react';
import EmailField from './EmailField.jsx';
import PasswordField from './PasswordField.jsx';
import Requisition from './../js/Requisition.js';
const R = Requisition;

const ForgotPasswordLink = props => <div className="popup_forgot"><a href="#">Forgot Password ?</a></div>;
const LoginLink = props => <a onClick={props.onClick} href="javascript:{}" className="ms_btn" target="_blank">login now</a>;
const RegisterLink = props => <p>Don't Have An Account? <a href="#myModal" data-toggle="modal" className="ms_modal1 hideCurrentModel">register here</a></p>
const ErrorMessage = props => props.error ? <div className="text-warning" role="alert">{props.error}</div> : null;

class LoginForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = { email: '', password: '' };
        this.handleEmail = this.handleEmail.bind(this);
        this.handlePassword = this.handlePassword.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    handleEmail(event) { 
        this.setState({ email: event.target.value, error: null }); 
    }
    handlePassword(event) { 
        this.setState({ password: event.target.value, error: null }); 
    }
    handleSubmit(event) { 
        if (this.validateForm()) {
            R.login(this.state.email, this.state.password, (errorCode, errorThrown) => {
                if (errorCode == 401)
                    this.setState({ error: "Invalid email or password" });
                else
                    this.setState({ error: "Unkown error of code " + errorCode + "." });
            }, () => this.setState({ error: null }));
        }
    }
    validateForm() {
        const form = document.getElementById('loginForm');
        form.classList.add('was-validated');
        return form.checkValidity();
    }
    render() {
        return (
            <form className="ms_register_form" id="loginForm" noValidate>
                <h2>login / Sign in</h2>
                <ErrorMessage error={this.state.error} />
                <EmailField value={this.state.email} onChange={this.handleEmail} errorMessage="Invalid email. Valid example: john@gmail.com"/>
                <PasswordField value={this.state.password} onChange={this.handlePassword} errorMessage="The password is required"/>
                <ForgotPasswordLink />
                <LoginLink onClick={this.handleSubmit}/>
            </form>
        );
    }
};

const RegisterImage = props => <div className="ms_register_img"> <img src="images/register_img.png" alt="" className="img-fluid" /> </div>;
const CloseButton = props => <button type="button" className="close" data-dismiss="modal"> <i className="fa_icon form_close" /> </button>;
    
class LoginModal extends React.Component {
    render() {
        return (
            <div id="myModal1" className="modal centered-modal" role="dialog">
                <div className="modal-dialog login_dialog">
                    // Modal content
                    <div className="modal-content">
                        <CloseButton />
                        <div className="modal-body">
                            <RegisterImage />
                            <LoginForm />
                        </div>
                    </div>
                </div>
            </div>
       );
   }
};

export default LoginModal;
