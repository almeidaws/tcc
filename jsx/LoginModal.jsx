class EmailField extends React.Component {
    render() {
        return (
            <div className="form-group">
                <input type="text" placeholder="Enter Your Email" className="form-control" data-rule="required|email"/>
                <span className="form_icon">
                    <i className="fa_icon form-envelope" aria-hidden="true" />
                </span>
            </div>
        );
    }
};

class PasswordField extends React.Component {
    render() {
        return (
            <div className="form-group">
                <input type="password" placeholder="Enter Password" className="form-control" />
                <span className="form_icon">
                    <i className="fa_icon form-lock" aria-hidden="true" />
                </span>
            </div>
        );
    }
};

const ForgotPasswordLink = props => (
    <div className="popup_forgot">
        <a href="#">Forgot Password ?</a>
    </div>
);

class LoginForm extends React.Component {
    render() {
        return (
            <form className="ms_register_form" id="loginForm">
                <h2>login / Sign in</h2>
                <EmailField />
                <PasswordField />
                <ForgotPasswordLink />
                <a href="profile.html" className="ms_btn" target="_blank">login now</a>
                <p>Don't Have An Account? <a href="#myModal" data-toggle="modal" className="ms_modal1 hideCurrentModel">register here</a></p>
            </form>
        );
    }
};

const RegisterImage = props => (
    <div className="ms_register_img">
        <img src="images/register_img.png" alt="" className="img-fluid" />
    </div>
);

const CloseButton = props => (
    <button type="button" className="close" data-dismiss="modal">
        <i className="fa_icon form_close" />
    </button>
);
    
class LoginModal extends React.Component {
    render() {
        return (
            <div id="myModal1" className="modal centered-modal" role="dialog">
                <div className="modal-dialog login_dialog">
                    // Modal content
                    <div className="modal-content">
                        <CloseButton />
                        <div className="modal-body">
                            <RegisterImage />
                            <LoginForm />
                        </div>
                    </div>
                </div>
            </div>
       );
   }
};

ReactDOM.render(<LoginModal />, document.getElementById('loginModal'));
