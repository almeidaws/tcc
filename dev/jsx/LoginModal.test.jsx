import React from 'react';
import LoginModal from './LoginModal.jsx';
import renderer from 'react-test-renderer';

it('<LoginModal /> should render static content', () => {
    const component = renderer.create(<LoginModal />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
