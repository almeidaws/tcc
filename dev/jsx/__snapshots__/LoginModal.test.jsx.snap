// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginModal /> should render static content 1`] = `
<div
  className="modal centered-modal"
  id="myModal1"
  role="dialog"
>
  <div
    className="modal-dialog login_dialog"
  >
    // Modal content
    <div
      className="modal-content"
    >
      <button
        className="close"
        data-dismiss="modal"
        type="button"
      >
         
        <i
          className="fa_icon form_close"
        />
         
      </button>
      <div
        className="modal-body"
      >
        <div
          className="ms_register_img"
        >
           
          <img
            alt=""
            className="img-fluid"
            src="images/register_img.png"
          />
           
        </div>
        <form
          className="ms_register_form"
          id="loginForm"
          noValidate={true}
        >
          <h2>
            login / Sign in
          </h2>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              required={true}
              type="email"
              value=""
            />
            <span
              className="form_icon"
            >
              <i
                aria-hidden="true"
                className="fa_icon form-envelope"
              />
            </span>
            <div
              className="invalid-feedback text-warning"
            >
              Invalid email. Valid example: john@gmail.com
            </div>
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
            <span
              className="form_icon"
            >
              <i
                aria-hidden="true"
                className="fa_icon form-lock"
              />
            </span>
            <div
              className="invalid-feedback text-warning"
            >
              The password is required
            </div>
          </div>
          <div
            className="popup_forgot"
          >
            <a
              href="#"
            >
              Forgot Password ?
            </a>
          </div>
          <a
            className="ms_btn"
            href="javascript:{}"
            onClick={[Function]}
            target="_blank"
          >
            login now
          </a>
        </form>
      </div>
    </div>
  </div>
</div>
`;
